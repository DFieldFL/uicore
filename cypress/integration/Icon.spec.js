/// <reference types="Cypress" />

context('<Icon/> Component', () => {
  before(() => {
    cy.visit('/icon')
    cy.hideStickyHeader()
  })

  after(() => {
    cy.showStickyHeader()
  })

  it('snapshots for <Icon/> with colors', () => {
    cy.get('#color-icons').toMatchSnapshot()
    cy.get('#color-icons').toMatchImageSnapshot()
  })

  it('snapshots for <Icon/> with styled props', () => {
    cy.get('#icon-styled-props').toMatchSnapshot()
    cy.get('#icon-styled-props').toMatchImageSnapshot()
  })
})
