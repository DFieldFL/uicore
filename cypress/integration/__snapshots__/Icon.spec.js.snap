exports[`<Icon/> Component > snapshots for <Icon/> with colors #0`] = `
<div
  id="color-icons"
  class="Layout---horizontal--3rmQ Layout---layout-spacing-small--2ZKb StyledProps---main--1Wws"
>
  <span class="bp3-icon StyledProps---main--1Wws"
    ><svg viewBox="0 0 14 13" width="64" height="64">
      <path d="M7 12.9L9.6 5H4.4z" fill="#e24329"></path>
      <path d="M7 12.9L4.4 5H.8z" fill="#fc6d26"></path>
      <path d="M.8 5L0 7.4c0 .2 0 .5.2.6L7 12.9z" fill="#fca326"></path>
      <path d="M.8 5h3.6L2.9.2c-.1-.2-.5-.2-.5 0z" fill="#e24329"></path>
      <path d="M7 12.9L9.6 5h3.6z" fill="#fc6d26"></path>
      <path d="M13.2 5l.8 2.4c0 .2 0 .5-.2.6L7 12.9z" fill="#fca326"></path>
      <path
        d="M13.2 5H9.6L11.2.2c.1-.2.4-.2.5 0z"
        fill="#e24329"
      ></path></svg></span
  ><span class="bp3-icon StyledProps---main--1Wws"
    ><svg viewBox="0 0 14 14" width="64" height="64">
      <path
        d="M2.2 12.5l.9-.2c.4.5.3.7.7 1 .6.5 1.4.3 1.7-.1.3-.5.3-.9.2-1.3l1-.2.1 1.1 5.9-1.1-.4-4.3h-.2L11.5.5.9 1.7z"
        fill="#fff"
      ></path>
      <path
        d="M11.3 9.9l-1.1-.8-.9 1.6-1.1-.3-1 1.5v.5l5.3-1-.3-3.3zM6.4 8.5l.8-.1c.1.1.2.1.4.1.3.1.6.1 1-.1L9 8l3.5-.6.4 4.3L7 12.8zm6.4-1.6l-.3.1-.7-6.8L.6 1.5 2 12.7l1.3-.2c-.1-.1-.3-.3-.5-.6-.4-.3-.2-.9 0-1.2.3-.6 1.9-1.3 1.8-2.3 0-.3-.1-.8-.4-1.1v.2s-.2 0-.3-.3c0 0-.1-.1-.1-.2v.3s-.2-.2-.2-.5c-.1.1-.1.3-.1.3s-.1-.4-.1-.6-.2-.5-.2-1.4c.3.2 1.1.2 1.4-.3.1-.1.2-.5 0-1.3-.2-.4-.5-1.2-.6-1.4 0 .2.1.7.2.9.2.7.2.9.1 1.3 0 .2-.2.4-.6.6s-.9-.3-1-.3c-.4-.3-.7-.8-.7-1.1s.2-.4.3-.7c-.2.1-.3.2-.3.2l.4-.4.2-.2h-.3s.2-.1.5-.2h-.4s.6-.2 1-.4c.3-.1.6-.1.8.1.2.3.4.5.9.6.3-.1.4-.2.7-.3.3-.3.6-.4.6-.4s-.2.1-.2.3l.4-.2s-.1 0-.2.2c.3-.1.5-.2.5-.2s-.1.1-.1.2h.6c.7 0 .9-.8 1.2-.9s.4-.2 1 .4c.5.5.9 1.4.7 1.6s-.5-.1-.9-.5c-.1-.2-.3-.5-.3-.9-.1-.3-.3-.5-.3-.5s.1.3.1.5c0 .1 0 .7.2 1 0 0 0 .2-.1.2-.2-.2-.7-.4-.8-.5.3.3 1 .8 1.3 1.4.3.5.1 1 .2 1.1 0 0 .6.7.7 1 .2.6 0 1.2-.2 1.5l-.7.2c-.1 0-.2 0-.2-.1s.1-.3.1-.3v-.1c-.2.3-.5.5-.8.7-.4.2-.8.2-1.1.1-.7-.3-1.5-.8-1.6-.9v.1c.2.2.6.6 1.1.9l-.9.1.4 3.4c-.2 0-.2 0-.4.1-.2-.7-.5-1.1-.9-1.3-.3-.2-.8-.3-1.3-.2.3 0 .7 0 1.1.3.4.2.7.8.8 1.2.1.5.2 1-.1 1.5-.3.4-1.1.6-1.7.1.2.3.4.5.7.5.5.1.9 0 1.2-.3s.4-.8.4-1.4l.2-.1.1 1 6.7-.8zM8.7 4.1v.2c.1.2.2.3.3.4h.4v-.1c0-.2 0-.6-.4-.7-.2-.1-.4 0-.4 0 .2.1.2.1.1.2zM9.9 6h-.5c-.3 0-.7.2-.8.2-.1.1-.1.3 0 .4.4.2.6.4.9.3.2 0 .3-.3.4-.5s.1-.3 0-.4zM7 4.3c.1-.1-.5-.2-.9.1-.3.2-.3.7 0 .9 0 0 .1 0 .1.1.1 0 .2-.1.3-.1.2-.1.4-.1.5-.1 0-.1.1-.2.1-.5-.1-.3-.3-.2-.1-.4z"
        fill="#7749a3"
      ></path></svg></span
  ><span class="bp3-icon StyledProps---main--1Wws"
    ><svg viewBox="0 0 14 14" width="64" height="64">
      <path
        d="M13.6 6.4l-6-6C7.3.1 6.7.1 6.4.4L5.1 1.6l1.6 1.6c.4-.1.8 0 1.1.3s.4.7.2 1.1l1.5 1.5c.4-.1.8 0 1.1.2.4.4.4 1.1 0 1.5s-1.1.4-1.5 0c-.3-.3-.3-.7-.2-1.1L7.5 5.3V9c.1 0 .2.1.3.2.4.4.4 1.1 0 1.5s-1.1.4-1.5 0-.4-1.1 0-1.5c.1-.1.2-.2.3-.2V5.2c-.1-.1-.2-.1-.3-.2-.3-.4-.4-.8-.2-1.2L4.5 2.3.4 6.4c-.3.3-.3.9 0 1.2l6 6c.3.3.9.3 1.2 0l6-6c.3-.3.3-.9 0-1.2z"
        fill="#f42534"
      ></path></svg></span
  ><span class="bp3-icon StyledProps---main--1Wws"
    ><svg viewBox="0 0 14 12" width="64" height="64">
      <mask
        id="service-gcp_svg__a"
        height="8.5"
        maskUnits="userSpaceOnUse"
        width="9.1"
        x="0"
        y="0"
      >
        <path d="M0 0h9.1v8.5H0z" fill="#fff"></path>
      </mask>
      <path d="M4.5 11L2 6.5 4.5 2h5L12 6.5 9.5 11z" fill="#cac9c2"></path>
      <path
        d="M7 7.8C5.9 7.8 5.1 7 5.1 6S5.9 4.1 7 4.1c1 0 1.9.9 1.9 1.9S8 7.8 7 7.8"
        fill="#fff"
      ></path>
      <path
        d="M10.8.5c-.1-.3-.5-.5-.8-.5H4c-.3 0-.7.2-.9.5L1.4 3.4c0 .2 0 .4.1.5l1.2 2 2.1-3.6h6.8c.2 0 .3.1.5.2z"
        fill="#5072aa"
      ></path>
      <path d="M1.4 3.3v.1c0 .2 0 .4.1.5l1.2 2 .6-1.1z" fill="#4d6796"></path>
      <path
        d="M4.1 1.5c-.2 0-.4-.2-.4-.4s.2-.4.4-.4.4.2.4.4c0 .3-.2.4-.4.4M10.2 1.2c0 .2-.2.3-.3.3s-.4-.1-.4-.3.2-.3.3-.3.4.1.4.3"
        fill="#fff"
      ></path>
      <path
        d="M13.9 5.5l-1.8-3c-.2-.1-.3-.2-.5-.2H9.1L11.2 6l-3.3 5.7-.3.3H10c.3 0 .7-.2.8-.5l3-5c.2-.3.2-.7.1-1"
        fill="#cc4232"
      ></path>
      <path
        d="M12.2 2.7l-.1-.1c-.2-.1-.3-.2-.5-.2H9.1l.6 1.1z"
        fill="#b23f33"
      ></path>
      <path
        d="M9.8 11.1c-.2 0-.4-.2-.4-.4s.2-.4.4-.4.4.2.4.4-.2.4-.4.4M12.7 6.3c-.2 0-.4-.2-.4-.4s.2-.4.4-.4.4.2.4.4c0 .3-.2.4-.4.4"
        fill="#fff"
      ></path>
      <path
        d="M4.9 6.1L1.5.5C1.4.4 1.4.2 1.4 0L.1 2.1c-.1.3-.1.7 0 .9l3 5c.2.3.6.5.9.5h3.6c.1-.1.3-.2.3-.3l1.2-2.1z"
        fill="#ecb02a"
        mask="url(#service-gcp_svg__a)"
        transform="translate(0 3.421)"
      ></path>
      <path d="M7.4 11.9h.2c.1-.1.3-.2.3-.3l1.2-2.1H7.8z" fill="#c79732"></path>
      <g fill="#fff">
        <path
          d="M4.1 11.1c-.2 0-.4-.2-.4-.4s.2-.4.4-.4.4.2.4.4-.2.4-.4.4M1.2 6.3C1 6.3.9 6.2.9 6s.2-.4.4-.4.4.2.4.4c-.1.2-.3.3-.5.3"
        ></path>
      </g></svg></span
  ><span class="bp3-icon StyledProps---main--1Wws"
    ><svg viewBox="0 0 10 14" width="64" height="64">
      <path
        d="M9.9 6.8c0 2.7-2.2 5-4.8 5-2.7 0-4.8-2.2-4.8-5 0-2.7 2.2-5 4.8-5s4.8 2.2 4.8 5"
        fill="#d33833"
      ></path>
      <path
        d="M.4 8S0 2.8 4.8 2.7l-.3-.6-2.6.9-.8.8-.6 1.3-.4 1.4.1 1"
        fill="#ef3d3a"
      ></path>
      <path
        d="M1.7 3.4C.9 4.3.3 5.5.3 6.8s.5 2.5 1.4 3.4 2 1.4 3.3 1.4 2.5-.5 3.3-1.4c.9-.9 1.4-2.1 1.4-3.4s-.5-2.5-1.4-3.4C7.5 2.5 6.3 2 5 2s-2.4.5-3.3 1.4zm-.2 7C.6 9.5 0 8.2 0 6.8s.6-2.7 1.5-3.6S3.6 1.7 5 1.7s2.6.6 3.5 1.5c1 .9 1.5 2.2 1.5 3.6s-.6 2.7-1.5 3.6-2.2 1.5-3.5 1.5c-1.4 0-2.6-.6-3.5-1.5z"
        fill="#231f20"
      ></path>
      <path
        d="M7.1 6.8l-.8.1-1 .1H4.1l-.5-.1-.5-.5-.3-.9-.1-.2-.4-.2-.3-.4-.1-.6.2-.5.5-.2.4.2.1.4h.2l.1-.1-.1-.5v-.5l.1-.7v-.4l.3-.5.6-.4 1-.4 1.1.2 1 .7.6.5.3.5.1 1.3-.2 1.1L7.7 6l-.4.5"
        fill="#f0d6b7"
      ></path>
      <path
        d="M6.4 10l-2.6.1v.4l.2 1.6-.1.1-1.8-.6-.1-.2-.3-2.1L1.3 8l-.1-.3 1.5-1 .5-.2.4.5.4.3.4.1.2.1.2 1 .2.2.4-.1-.4.5 1.6.8z"
        fill="#335061"
      ></path>
      <path
        d="M2.1 3.5l.5-.2.4.2.1.5h.2l.1-.2-.1-.5.1-1-.1-.6.3-.4.7-.6-.2-.2-1 .5-.4.3-.2.5-.4.5-.1.6z"
        fill="#6d6b6d"
      ></path>
      <path
        d="M2.8 1.8s.3-.7 1.4-1 .1-.2.1-.2L3.1 1l-.5.5-.2.3zM2.3 3.4s-.4-1.3 1.1-1.5l-.1-.2-1 .3-.3.9.1.6z"
        fill="#dcd9d8"
      ></path>
      <g fill="#f7e4cd">
        <path
          d="M2.9 5.1l.2-.2s.1 0 .1.1.1 1.3.9 1.9c.1.1-.6 0-.6 0l-.6-1M6.3 4.8s0-.6.2-.5c.2 0 .2.2.2.2s-.5.2-.4.3M7.8 2.7s-.3.1-.3.3c0 .3.3.1.4 0M5.5 2.8s-.4.1-.4.3c0 .3.5.2.6.1M3 4s-.7-.4-.8 0c0 .3-.2.6.2 1.1L2.1 5l-.2-.6-.1-.5.4-.4h.5l.3.2zM3.4 2.8S3.7 1.2 5.3.9c1.3-.3 2 .1 2.3.4 0 0-1.2-1.4-2.3-1-1.2.4-2 1.1-1.9 1.6zM7.7 1.4s-.5 0-.6.5V2s.4-.5.7-.2"
        ></path>
        <path
          d="M5.3 2s-.1-.7-.7-.3c-.4.3-.4.7-.3.8s.1.2.1.1c.1-.1 0-.4.2-.5s.6-.2.7-.1"
        ></path>
      </g>
      <path
        d="M3.6 7l-1.8.8s.7 2.9.4 3.8l-.3-.1v-1.1l-.5-2.1-.2-.6L3 6.5zM3.8 8.6l.2.3V10h-.3v-.9c0-.1 0-.4.1-.5M3.8 10.2H3l.2.2.6.1"
        fill="#49728b"
      ></path>
      <path
        d="M6.6 10h.7l.2 1.7-.7.1zM6.8 10l1-.1.4-1.1.4-1.6-.8-.8-.2-.1-.4.4v1.7z"
        fill="#335061"
      ></path>
      <path
        d="M7.2 9.9l-.6.1.1.5c.2.1.6-.2.6-.2M7.3 6.6l1.3 1v-.4l-1-.9z"
        fill="#49728b"
      ></path>
      <path
        d="M4.4 13.7L4 12.1 3.8 11v-.8l1.7-.2h1.1l-.1 2 .2 1.5v.3l-1.4.1z"
        fill="#fff"
      ></path>
      <path
        d="M6.4 10s-.1 1.9.2 3.3c0 0-.6.4-1.4.4l1.6-.1.2-.1-.2-3-.2-.6"
        fill="#dcd9d8"
      ></path>
      <path
        d="M7.5 11.5l.7-.2 1.4-.1.2-.6-.4-1.1H9l-.6.2-.6.3-.3-.1-.2.1"
        fill="#fff"
      ></path>
      <path
        d="M7.5 11.2s.5-.2.6-.2l-.2-1 .2-.1s.2 1 .2 1.1c0 0 1 .1 1.1.1 0 0 .2-.4.2-.9l.2.6v.3l-.3.4-.3.1h-.6l-.2-.2-.6.1-.2.1"
        fill="#dcd9d8"
      ></path>
      <path d="M6.7 9.9l-.4-1-.4-.7S6 8 6.1 8h.4l.5.1v.7z" fill="#fff"></path>
      <path
        d="M6.8 9.5s-.5-1-.5-1.1c0 0 .1-.2.2-.2.1.1.4.2.4.2v-.3l-.6-.2-.4.1.7 1.7h.1"
        fill="#dcd9d8"
      ></path>
      <path d="M4.6 7.1L4.1 7l-.5-.1V7l.2.3.7.3" fill="#fff"></path>
      <path d="M3.7 7.1s.6.2.8.2v.2L4 7.4z" fill="#dcd9d8"></path>
      <path
        d="M7.5 8c-.3 0-.6 0-.9-.1v-.3c.1 0 .2 0 .3-.1-.1 0-.2-.1-.3 0 0-.1 0-.1-.1-.2.2 0 .7-.4.9-.3.1.1.2.4.2.6s-.1.3-.1.4M6 7.5s0 .1 0 0c-.1.1-.3.1-.4.2.2 0 .3 0 .4.1V8c-.2.1-.3.3-.6.4-.1.1-.4.2-.5.2s-.1-.1-.1-.2c-.1-.1-.2-.4-.2-.6 0-.3 0-.8.3-.7.2 0 .5.2.7.3.1 0 .2.1.4.1"
        fill="#d33833"
      ></path>
      <path d="M6.2 7.9c0-.2-.1-.2 0-.3.3-.4.4.4 0 .3" fill="#d33833"></path>
      <path
        d="M6.8 8s-.1-.2 0-.2c.1-.1.2 0 .2-.1v-.3c0-.1.1-.1.2-.1s.4-.1.4 0l-.1-.4h-.3l-.7.4V7.9M4.8 8.6c0-.3 0-.6-.1-.9 0-.4.1-.4.5-.4.1 0 .4.1.4.1.1.2-.2.2.1.3.3.2.7 0 .6-.3h-.4c-.1-.1-.3-.2-.4-.3-.2-.1-.6-.2-.8-.1-.5.4 0 1.2.1 1.6"
        fill="#ef3d3a"
      ></path>
      <g fill="#231f20">
        <path
          d="M5.3 2c-.5-.1-.7.3-.9.6-.1 0-.1-.2 0-.3 0-.3.4-.6.7-.6.1.1.3.2.2.3M7.8 2.6c.1.2.2.5.4.7-.1.2-.7.4-.7 0 .1-.1.4 0 .5-.1-.2-.1-.3-.3-.2-.6M5.6 2.6c.1.2.1.4.3.6 0 0 .2.1.1.3l-.2.2c-.2 0-.8 0-.6-.3.2 0 .5.1.6 0-.1-.2-.3-.6-.2-.8M7.7 4.6c-.4.3-.8.5-1.4.5-.1-.1-.2-.4 0-.6.1.1 0 .3.2.3.3.1.7-.2 1-.3.1-.3 0-.3-.1-.5-.4-.3-.8-.7-.7-1.2.1 0 .1.2.1.2.1.3.5.8.8 1.1.1.1.2.1.2.2s-.1.2-.1.3M2.9 4.4c-.1-.1-.1-.4-.3-.4s-.1.4-.1.6c-.1-.1-.2-.5-.1-.6-.1-.1-.1 0-.2.1.1-.6.8-.3.7.3M7.9 4.9c-.2.3-.4.6-.9.7v-.3c.4-.1.6-.3.9-.4M5.5 5.1c.3.2.9.2 1.3.2v.3c-.5 0-1.1-.1-1.3-.5M5.5 5.4c.2.5 1 .5 1.6.5-.1 0-.1.1-.2.1-.2.1-.8.1-1 0s-.3-.3-.4-.4-.3-.2 0-.2"
        ></path>
      </g>
      <path
        d="M7.6 8.4c-.2.4-.5.9-.8 1.3.2-.4.2-1 .2-1.5.3-.1.5 0 .6.2"
        fill="#81b0c4"
      ></path>
      <path
        d="M9 10c-.3.1-.5.3-.8.3.2-.2.5-.3.8-.3M9.1 10.4c-.2 0-.5.1-.7 0 .1-.1.5 0 .7 0M9.2 10.8h-.8c.1-.2.7-.1.8 0"
        fill="#231f20"
      ></path>
      <path
        d="M7.2 11.9c0 .3.2.7.2 1-.1 0-.2.1-.4.1 0-.3-.1-.8 0-1.1-.1 0 .1-.1.2 0"
        fill="#dcd9d8"
      ></path>
      <path
        d="M6.7 7l-.3.3c-.3 0-.4 0-.6-.2.2.1.6-.1.9-.1"
        fill="#f0d6b7"
      ></path>
      <path
        d="M5.2 9.1c.1-.4.4-.5.7-.7.3.4.5.9.7 1.4-.5-.3-1-.5-1.4-.7"
        fill="#81b0c4"
      ></path>
      <path
        d="M6.8 11.9V13c.2 0 .3 0 .4-.1 0-.4-.1-.7-.2-1 .1-.1-.1 0-.2 0zm-3-1.7c.1 1.2.3 2.2.7 3.2.8.2 1.7.2 2.3 0-.1-.6-.1-1.3-.1-1.9-.1-.5 0-.9-.1-1.4-.8-.1-2 0-2.8.1zm3-.1c0 .5 0 1 .1 1.5.2 0 .3 0 .5-.1-.1-.5-.1-1-.2-1.5-.2.1-.3.1-.4.1zm.9 0h-.3c0 .4.1.9.2 1.3.1 0 .2-.1.3-.1.1-.4 0-.9-.2-1.2zm1.5 1.3c.3-.1.5-.4.4-.8-.1-.2-.2-.7-.3-.9-.1-.1-.3-.3-.4-.2-.3.2-.7.2-.9.4.1.3.1.8.2 1.2.3 0 .7-.1 1 0-.2.1-.4.1-.6.2.1.2.4.2.6.1zM6.5 9.7c-.2-.5-.4-1-.7-1.4-.2.2-.6.4-.6.8.4.2.9.4 1.3.6zM7 8.2c0 .5 0 1.1-.2 1.5.3-.4.5-.8.8-1.3-.1-.2-.3-.3-.6-.2zM6.5 8c-.1 0-.2.1-.4.1l-.1.1c.3.4.5 1 .7 1.4.1-.4.1-.9.2-1.4-.2 0-.3-.2-.4-.2zm-.4-.5v.3c.5.2.4-.6 0-.3zm-.5-.1c-.2-.2-.5-.3-.7-.3-.3-.1-.3.4-.3.7 0 .2.1.5.2.6 0 .1 0 .1.1.2.1 0 .5-.1.6-.2.2-.1.3-.3.5-.4v-.2c-.1-.1-.2-.1-.4-.1.1-.1.3-.1.4-.1v-.1c-.2 0-.3-.1-.4-.1zM3.8 7c-.2.2.4.4.6.4 0-.1.1-.2 0-.3-.2 0-.5 0-.6-.1zm1.9.1c.2.2.3.2.6.2.2-.1.3-.2.4-.3-.3 0-.7.2-1 .1zm1.9.5c0-.2-.1-.6-.2-.6-.2-.1-.7.3-.8.4 0 .1.1.1.1.2H7c-.1 0-.2 0-.3.1V8h.8c0-.1.1-.2.1-.4zm-4.1-.4c-.1 0-.4-.5-.4-.5-.6.3-1.2.7-1.7 1.1.5 1.1.7 2.3.7 3.6.6.2 1 .6 1.8.6-.1-.6-.2-1.2-.2-1.8-.2-.1-.5 0-.6 0-.1-.2.2-.1.2-.2s-.2-.1-.1-.3c.1.1.2.2.4.2.1-.3 0-.8 0-1.1 0 0 0-.3.1-.2.1 0 0 .6 0 .8s0 .4.1.6c.7-.1 1.5-.2 2.3-.2-.2-.1-.4-.1-.6-.3-.1 0-.5-.2-.6-.3 0-.2.2-.3.2-.5-.4.2-.4-.1-.5-.4s-.1-.5-.2-.6c-.3-.2-.6-.3-.9-.5zm3.7-.4c.5-.2.6.9.4 1.3 0 .1.1.2.2.3l-.9 1.5c.2-.1.5 0 .8-.1.1 0 .2-.3.2-.4.2-.5.4-1.1.5-1.5 0-.1.1-.3.1-.4 0-.2-.2-.3-.4-.4-.2-.3-.3-.5-.5-.7-.1.2-.3.3-.4.4zM2.3 2.2c-.2.3-.2.8-.1 1.2.4-.3 1 0 1 .5.2 0 .1-.3 0-.4-.1-.5.2-1.1 0-1.6-.3-.1-.6.1-.9.3zM4.1.7c-.6.1-1.3.5-1.5 1 .2 0 .3-.1.5-.1h.2c.2 0 .3-.4.4-.5L4 .8c.1 0 .2 0 .1-.1zM7 .8C6.4.5 5.4.2 4.8.6c-.5.2-1.2.6-1.4 1.1.2.5-.1 1-.1 1.5 0 .3.1.5.1.8 0 .1-.2.1-.4.1 0-.3-.1-.6-.4-.6-.4 0-.6.3-.6.6 0 .4.3 1 .7 1 .2 0 .2-.2.4-.2.1.2-.2.3-.2.4v.3c.1.3.3.8.4 1 .3.3.8.4 1.2.4.1-.2.4-.2.6-.1-.2-.1-.4-.4-.6-.6S4 5.8 4 5.5c.4.6.7 1.1 1.5 1.3.6.2 1.2-.1 1.7-.4.2-.1.3-.3.4-.5.5-.7.7-1.7.7-2.6 0-.4 0-.8-.2-1-.2-.3-.6-.6-.9-.3 0-.3.2-.4.6-.3-.3-.3-.5-.7-.8-.9zm1 8.9c.4-.2 1.3-.6 1.6 0 .1.2.2.6.3.8.1.3-.1 1-.4 1.1s-.6.1-1 0c0 0-.1-.1-.1-.2-.3 0-.5 0-.7.1 0 .2-.1.2-.2.3-.1.3.2.8.1 1.1 0 .2-.3.3-.5.3v.3c0 .2-.3.3-.5.3-.7.2-1.6.2-2.3 0-.2-.4-.3-1-.5-1.4-.6.1-1.1-.3-1.6-.5-.2-.1-.4-.1-.4-.2-.1-.1 0-.4 0-.6 0-.6-.1-1.1-.2-1.7-.1-.4-.2-.6-.3-.9-.1-.2-.2-.5-.2-.7-.1-.4.2-.4.5-.5.3-.3.5-.4.9-.6.1-.1.4-.2.5-.3 0-.2-.2-.4-.2-.5-.1-.2-.2-.3-.2-.5-.3 0-.5-.2-.6-.4-.2-.3-.4-.9-.2-1.4 0 0 .1-.1.1-.2V3c-.1-.5.1-1 .5-1.2.2-.7.7-.9 1.3-1.2.2-.1.4-.2.7-.3.8-.3 2.1-.3 2.8.3.3.2.8.7.9 1 .4.9.4 2.4.1 3.5 0 .1-.1.4-.2.5 0 .2-.2.4-.2.6 0 .1.4.4.5.5.2.1.4.3.5.5 0 .2-.1.5-.1.7-.3.6-.5 1.2-.7 1.8z"
        fill="#231f20"
      ></path>
      <path
        d="M5 5.2s.2-.1.3 0c0 0-.2 0-.2.4H5c.1 0 0-.3 0-.4"
        fill="#f7e4cd"
      ></path>
      <path
        d="M6.7 8.4c0 .1 0 .1-.1.1s-.1 0-.1-.1 0-.1.2 0c-.1-.1 0-.1 0 0M6.8 8.9c0 .1 0 .1-.1.1s-.1 0-.1-.1 0-.1.1-.1c0 0 .1 0 .1.1"
        fill="#1d1919"
      ></path></svg></span
  ><span class="bp3-icon StyledProps---main--1Wws"
    ><svg viewBox="0 0 14 14" width="64" height="64">
      <g fill="#e01e5a">
        <path
          d="M3 8.8c0 .8-.7 1.5-1.5 1.5S0 9.6 0 8.8s.7-1.4 1.5-1.4H3zM3.7 8.8c0-.8.7-1.5 1.5-1.5s1.4.7 1.4 1.5v3.7c0 .8-.6 1.5-1.4 1.5s-1.5-.7-1.5-1.5z"
        ></path>
      </g>
      <g fill="#36c5f0">
        <path
          d="M5.2 3c-.8 0-1.5-.7-1.5-1.5S4.4 0 5.2 0s1.5.7 1.5 1.5V3zM5.2 3.7c.8 0 1.5.7 1.5 1.5S6 6.6 5.2 6.6H1.5C.7 6.6 0 6 0 5.2s.7-1.5 1.5-1.5z"
        ></path>
      </g>
      <g fill="#2eb67d">
        <path
          d="M11 5.2c0-.8.7-1.5 1.5-1.5s1.5.7 1.5 1.5-.7 1.5-1.5 1.5H11zM10.3 5.2c0 .8-.7 1.5-1.5 1.5C8 6.6 7.4 6 7.4 5.2V1.5C7.4.7 8 0 8.8 0s1.5.7 1.5 1.5z"
        ></path>
      </g>
      <g fill="#ecb22e">
        <path
          d="M8.8 11c.8 0 1.5.7 1.5 1.5S9.6 14 8.8 14s-1.5-.7-1.5-1.5V11zM8.8 10.3c-.8 0-1.5-.7-1.5-1.5.1-.8.7-1.4 1.5-1.4h3.7c.8 0 1.5.6 1.5 1.4s-.7 1.5-1.5 1.5z"
        ></path>
      </g></svg
  ></span>
</div>
`;

exports[`<Icon/> Component > snapshots for <Icon/> with styled props #0`] = `
<span
  id="icon-styled-props"
  icon="heart"
  class="bp3-icon bp3-icon-heart StyledProps---main--1Wws StyledProps---color--8PuS StyledProps---color-blue500--3HtV StyledProps---padding-small--3f0A StyledProps---border--uz_- StyledProps---border-color-blue400--Ophv"
  ><svg data-icon="heart" width="64" height="64" viewBox="0 0 20 20">
    <desc>heart</desc>
    <path
      d="M20 6.25C20 3.35 17.65 1 14.75 1c-1.02 0-1.95.31-2.75.82v-.04c-.09.06-.17.12-.26.19-.04.03-.09.06-.14.1-.68.51-1.24 1.18-1.6 1.96-.4-.86-1.04-1.57-1.8-2.1-.04-.02-.07-.05-.1-.08a7 7 0 00-.6-.33c-.13-.04-.23-.1-.35-.15-.05-.02-.1-.05-.15-.07v.02C6.45 1.13 5.87 1 5.25 1A5.25 5.25 0 000 6.25c0 .09.01.17.01.25H0c0 .06.01.12.02.18s.01.12.02.18C.13 7.89.44 9 1.07 10.17 2.23 12.33 4.1 14.11 7 16.53v.01c.9.75 1.89 1.55 3 2.46.71-.58 1.38-1.12 2-1.63 3.48-2.86 5.64-4.78 6.93-7.18.63-1.17.94-2.27 1.03-3.3.01-.07.01-.14.02-.21 0-.06.01-.11.02-.17h-.01c0-.09.01-.17.01-.26z"
      fill-rule="evenodd"
    ></path></svg
></span>
`;
